crates/chia-consensus/src/npc_result.rs

```diff
@@ -0,0 +1,93 @@
+use crate::allocator::make_allocator;
+use crate::consensus_constants::ConsensusConstants;
+use crate::gen::condition_tools::u64_to_bytes;
+use crate::gen::conditions::{
+    process_single_spend, validate_conditions, EmptyVisitor, ParseState, SpendBundleConditions,
+};
+use crate::gen::flags::MEMPOOL_MODE;
+use crate::gen::owned_conditions::OwnedSpendBundleConditions;
+use crate::gen::run_block_generator::subtract_cost;
+use crate::gen::validation_error::{ErrorCode, ValidationErr};
+use crate::multiprocess_validation::get_flags_for_height_and_constants;
+use chia_protocol::SpendBundle;
+use clvm_utils::{tree_hash_cached, TreeHash};
+use clvmr::allocator::{Allocator, NodePtr};
+use clvmr::chia_dialect::ChiaDialect;
+use clvmr::reduction::Reduction;
+use clvmr::run_program::run_program;
+use clvmr::serde::node_from_bytes;
+use std::collections::{HashMap, HashSet};
+// #[cfg(feature = "py-bindings")]
+// use chia_py_streamable_macro::{PyGetters, PyJsonDict, PyStreamable};
+use clvmr::chia_dialect::LIMIT_HEAP;
+
+// we may be able to remove this struct and just return a Rust native Result
+
+// #[cfg_attr(
+//     feature = "py-bindings",
+//     pyo3::pyclass(module = "chia_rs"),
+//     derive(PyJsonDict, PyStreamable, PyGetters),
+//     py_uppercase,
+//     py_pickle
+// )]
+// #[streamable]
+// pub struct NPCResult {
+//     error: Option<u16>,
+//     conds: Option<OwnedSpendBundleConditions>,
+// }
+
+pub fn get_name_puzzle_conditions(
```

Comment:
Eventually we'll want to use it for other purposes though right? The mempool param isn't taking up a lot of overhead

Reply:
Most of the suggestions I've made to simplify this was based on the assumption that we would use a separate function for full blocks.

The cost of making this function work both for mempool (i.e. single spend bundles) and full blocks is:
1. We would need to turn the spend bundle into a block before calling this function (in order to have a uniform interface). This function would then need to run said block generator function to get back to the individual spends. As it is now, we can skip that part.
3. We would need to support both pre- and post- hard fork (since blocks need to validate with the pre-hard fork rules). Transactions going into the mempool only need to be validated with the post-hard fork rules however.